src/ucp/core/ucp_worker.c

```diff
@@ -1318,7 +1318,71 @@ static void ucp_worker_init_cpu_atomics(ucp_worker_h worker)
     }
 }
 
-static void ucp_worker_init_device_atomics(ucp_worker_h worker)
+static int ucp_resources_is_device_shared(const ucp_tl_resource_desc_t *rsc1,
+                                          const ucp_tl_resource_desc_t *rsc2)
+{
+    return (rsc1->md_index == rsc2->md_index) &&
+           !strncmp(rsc1->tl_rsc.dev_name, rsc2->tl_rsc.dev_name,
```

Comment:
rsc1->dev_index == rsc2->dev_index

Reply:
by some reason this did not work, will debug